lib/Runtime/Library/JavascriptGeneratorFunction.cpp

```diff
@@ -86,6 +129,63 @@ namespace Js
         return generator;
     }
 
+    JavascriptGenerator* JavascriptAsyncFunction::CreateGenerator(Arguments& stackArgs)
+    {
+        ScriptContext* scriptContext = this->GetScriptContext();
+
+        // InterpreterStackFrame takes a pointer to the args, so copy them to the recycler heap
+        // and use that buffer for this InterpreterStackFrame.
+        Var* argsHeapCopy = RecyclerNewArray(scriptContext->GetRecycler(), Var, stackArgs.Info.Count);
+        js_memcpy_s(argsHeapCopy, sizeof(Var) * stackArgs.Info.Count, stackArgs.Values, sizeof(Var) * stackArgs.Info.Count);
+        Arguments heapArgs(stackArgs.Info, argsHeapCopy);
+
+        DynamicObject* prototype = scriptContext->GetLibrary()->CreateGeneratorConstructorPrototypeObject();
+        JavascriptGenerator* generator = scriptContext->GetLibrary()->CreateGenerator(heapArgs, this->GetGeneratorVirtualScriptFunction(), prototype);
+
+        return generator;
+    }
+
+    Var JavascriptGeneratorFunction::EntryAsyncFunctionImplementation(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+        ARGUMENTS(stackArgs, callInfo);
+
+        ScriptContext* scriptContext = function->GetScriptContext();
+        JavascriptLibrary* library = scriptContext->GetLibrary();
+
+        Var* argsHeapCopy = RecyclerNewArray(scriptContext->GetRecycler(), Var, stackArgs.Info.Count);
+        js_memcpy_s(argsHeapCopy, sizeof(Var) * stackArgs.Info.Count, stackArgs.Values, sizeof(Var) * stackArgs.Info.Count);
+        Arguments heapArgs(callInfo, argsHeapCopy);
```

Comment:
Are we getting two heap copies of the arguments since we're doing another heap allocation and copy above in CreateGenerator?  We only need one heap allocated copy of the arguments.

Please check.